### PR TITLE
feat: unify all release steps in a single pipeline

### PR DESCRIPTION
**Pull request by LucaLanziani**
The previous solution published the npm package before the go binary was built and attached to the release, causing the npm package to fail without error messages.

Adding all steps in a single action will ensure the right order and the npm package will work right after being published

### CODE REVIEW

File: .github/workflows/release-please.yaml [hunk 1]
```diff
@@ -3,37 +3,50 @@ on:
...
       - uses: google-github-actions/release-please-action@v3
         id: release
         with:
           command: manifest
-          token: ${{ steps.get_installation_token.outputs.token }}
 
-      # The logic below handles the npm publication
-      - uses: actions/checkout@v2
+## Build go binary and add it to the release
+      - uses: actions/checkout@v3
```
Comment (nearform-kenan) [inline .github/workflows/release-please.yaml:23]: please bump the version to `v4`

Comment (nearform-kenan) [inline .github/workflows/release-please.yaml:23]: please bump the version to `v4`

File: .github/workflows/release-please.yaml [hunk 2]
```diff
@@ -3,37 +3,50 @@ on:
...
+        run: git fetch --force --tags
+        if: ${{ steps.release.outputs.release_created }}
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version: stable
+        if: ${{ steps.release.outputs.release_created }}
+
+      - name: Build go binaries
+        uses: goreleaser/goreleaser-action@v4
```
Comment (nearform-kenan) [inline .github/workflows/release-please.yaml:39]: please bump the version to `v5`

File: .github/workflows/release-please.yaml [hunk 3]
```diff
@@ -3,37 +3,50 @@ on:
...
+          distribution: goreleaser
+          version: latest
+          args: release --clean
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         if: ${{ steps.release.outputs.release_created }}
 
-      # The logic below handles the npm publication:
+## Publish the release to npm
       - uses: actions/checkout@v2
```
Comment (nearform-kenan) [inline .github/workflows/release-please.yaml:49]: latest version is `v4`

**Review by nearform-kenan — CHANGES_REQUESTED** [review]

File: .github/workflows/release-please.yaml [hunk 4]
```diff
@@ -3,37 +3,50 @@ on:
...
 
-      # The logic below handles the npm publication
-      - uses: actions/checkout@v2
+## Build go binary and add it to the release
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+        if: ${{ steps.release.outputs.release_created }}
+
+      - name: fetch tags
```
Comment (zaremba-tomasz) [inline .github/workflows/release-please.yaml:28]: What's the point of this step?

**Review by zaremba-tomasz — APPROVED** [review]

**Review by nearform-kenan — APPROVED** [review]

**Pull request merged**